pkg/cmd/app.go

```diff
@@ -346,22 +310,49 @@ restart:
 
 	cRegistry := dcgmexporter.NewRegistry()
 
-	if dcgmexporter.IsDCGMExpXIDErrorsCountEnabled(cs.ExporterCounters) {
-		item, exists := fieldEntityGroupTypeSystemInfo.Get(dcgm.FE_GPU)
-		if !exists {
-			logrus.Fatalf("%s collector cannot be initialized", dcgmexporter.DCGMXIDErrorsCount.String())
-		}
+	enableDCGMExpXIDErrorsCountCollector(cs, fieldEntityGroupTypeSystemInfo, hostname, config, cRegistry)
 
-		xidCollector, err := dcgmexporter.NewXIDCollector(cs.ExporterCounters, hostname, config, item)
-		if err != nil {
-			logrus.Fatal(err)
-		}
+	enableDCGMExpClockEventsCount(cs, fieldEntityGroupTypeSystemInfo, hostname, config, cRegistry)
 
-		cRegistry.Register(xidCollector)
+	defer func() {
+		cRegistry.Cleanup()
+	}()
 
-		logrus.Infof("%s collector initialized", dcgmexporter.DCGMXIDErrorsCount.String())
+	ch := make(chan string, 10)
+
+	var wg sync.WaitGroup
+	stop := make(chan interface{})
+
+	wg.Add(1)
+	go pipeline.Run(ch, stop, &wg)
+
+	wg.Add(1)
+
+	server, cleanup, err := dcgmexporter.NewMetricsServer(config, ch, cRegistry)
```

Comment:
I see some change in the ordering, is that on purpose? Should this be before line 321?

Reply:
@rohit-arora-dev, Yes, the change in the ordering was made to move variable assignments closer to the first usage.